modules/fedimint-mint-client/src/lib.rs

```diff
@@ -722,35 +732,57 @@ impl ClientModule for MintClientModule {
 }
 
 impl MintClientModule {
-    /// Returns the number of held e-cash notes per denomination
-    pub async fn get_wallet_summary(&self, dbtx: &mut DatabaseTransaction<'_>) -> TieredSummary {
-        dbtx.find_by_prefix(&NoteKeyPrefix)
-            .await
-            .fold(
-                TieredSummary::default(),
-                |mut acc, (key, _note)| async move {
-                    acc.inc(key.amount, 1);
-                    acc
-                },
-            )
-            .await
+    async fn create_sufficient_input(
+        &self,
+        dbtx: &mut DatabaseTransaction<'_>,
+        operation_id: OperationId,
+        min_amount: Amount,
+    ) -> anyhow::Result<(Vec<ClientInput<MintInput, MintClientStateMachines>>, Amount)> {
+        if min_amount == Amount::ZERO {
+            return Ok((Vec::new(), Amount::ZERO));
+        }
+
+        let selected_notes = Self::select_notes(
+            dbtx,
+            &SelectNotesWithAtleastAmount,
+            min_amount,
+            self.cfg.fee_consensus.note_spend_abs,
+        )
+        .await?;
+
+        let selected_amount = selected_notes.total_amount();
+
+        for (amount, note) in selected_notes.iter_items() {
+            dbtx.remove_entry(&NoteKey {
+                amount,
+                nonce: note.nonce(),
+            })
+            .await;
+        }
+
+        let inputs = self
+            .create_input_from_notes(operation_id, selected_notes)
+            .await?;
+
+        assert!(!inputs.is_empty());
+
+        Ok((inputs, selected_amount))
```

Comment:
Instead of returning the selected amount, why not just sum the input amount where needed? Returning this tuple was very confusing to me at the call-site and is also redundant and thus a potential source of bugs.

Reply:
Done.